### PR TITLE
Automatically update dataset variables based on JSON files

### PR DESCRIPTION
**Pull request by mjaehn**
Fixes https://github.com/C2SM/c2sm.github.io/issues/129

### CODE REVIEW

**Comment by mjaehn** [comment]
@AnnikaLau @jonasjucker I would be overly greatful if one of you could approve after looking at the Datasets (which I showed in the Group Meeting today) as it would allow me to work further on the automatization. 🙏🏻

File: scripts/update_datasets.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                        variable_details[variable] = {}
+                    if scenario not in variable_details[variable]:
+                        variable_details[variable][scenario] = set()
+                    # Add resolution to the set of resolutions for this scenario
+                    variable_details[variable][scenario].add(temporal_resolution)
+        
+        first_iteration = True
+        for variable, scenarios_resolutions in variable_details.items():
+            markdown_content += process_cordex_data(variable, scenarios_resolutions, first_iteration)
+            first_iteration = False
```
Comment (jonasjucker) [inline scripts/update_datasets.py:39]: Could you split this function into two parts?
1. generate_markdow_for_cordex
2. generate_markdown_for_default

It is hard to find out what the function actually does.

**Review by jonasjucker — CHANGES_REQUESTED** [review]
Could you split some of the bigger functions into smaller pieces?
Otherwise LGTM

File: scripts/update_datasets.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        variable_info += f",\n  `{variable}`{{ title=\""
+    first_iteration_info = True
+    if dataset == 'cmip6-ng':
+        for resolution, grid in resolutions.items():
+            for grid_type, grid_values in grid.items():
+                if first_iteration_info:
+                    variable_info += f"{resolution} ({grid_type})"
+                    first_iteration_info = False
+                else:
+                    variable_info += f", {resolution} ({grid_type})"
```
[the remark this answers is not in the document]
Comment (mjaehn) [reply scripts/update_datasets.py]: Done in https://github.com/C2SM/c2sm.github.io/pull/133/commits/74edb1a37b3e78bd449ae33503e3bab71e1eca1c

File: scripts/update_datasets.py (hunk 1 above)
Comment (mjaehn) [reply scripts/update_datasets.py]: I have refactored this function in https://github.com/C2SM/c2sm.github.io/pull/133/commits/3f4e68cab5b200e176f6b76920ef8d035081375c and added docstrings in https://github.com/C2SM/c2sm.github.io/pull/133/commits/5dbf37d3dce81ef4e95e2db93c84822811c0617f. Should be clearer now.

**Review by jonasjucker — APPROVED** [review]

**Pull request merged**